File: B/wc.py
#!/usr/bin/python 
from pyspark import SparkContext
import sys
import os
import logging
import logging.handlers
import resource
import time
import subprocess, threading
from collections import defaultdict
from os.path import realpath
from operator import add
from helper import parseRecord, getCDFList, FIELD, HO_TYPE
    
if __name__ == "__main__":
    if len(sys.argv) < 3:
        print >> sys.stderr, "Usage: wc <master> <input dir> <log file>"
        exit(-1)

    sys.stdout = open('o.txt', 'w')
    master = sys.argv[1]
    input_dir = sys.argv[2]
    log_file = sys.argv[3]
    sys.stderr = open(log_file, 'w')
    
    #sc = SparkContext("local[" + numCores + "]" , "job", pyFiles=[realpath('helper.py')])
    sc = SparkContext(master , "job", pyFiles=[realpath('helper.py')])
    lines = sc.textFile(input_dir + '*')
    counts = lines.flatMap(lambda x: x.split(' ')).map(lambda x: (x, 1)).reduceByKey(add,64)
    output = counts.collect()
    

File: scripts/src/main/scala/wc.scala
import org.apache.spark.SparkContext
import org.apache.spark.SparkContext._
import org.apache.spark.scheduler.JobLogger
import org.apache.spark.scheduler

object WordCount {
  def main(args: Array[String]) {
    
    if (args.length < 3) {
      println("Usage: wc <master> <input dir> <log file>")
      exit()
    }

    val master = args(0)
    val input_dir = args(1)
    val log_file = args(2)

    val sc = new SparkContext(master, "job", "/homes/network/revtr/ujaved/incubator-spark",
      List("target/scala-2.9.3/simple-project_2.9.3-1.0.jar"))

    val listener = new JobLogger("ujaved")
    sc.addSparkListener(listener)

    println(System.getenv("SPARK_LOG_DIR"))

    val lines = sc.textFile(input_dir + "*")
    val counts = lines.flatMap(line => line.split(' ')).map(word => (word,1)).reduceByKey(_+_)
    val output = counts.collect().length
    println("count : %s".format(output))
  }
}
